tests/unit-tests/UnitTests/Services/TemplatedEmailTests.cs

```diff
@@ -70,5 +73,107 @@ public void Process_ShouldCallSendEmailAsync_ForTemplatedEmailType()
                 It.IsAny<string>(),
                 It.IsAny<Dictionary<string, dynamic>>()), Times.Once);
         }
+
+        [Fact]
+        public void Process_ShouldCallSendEmailAsync_WithCaseReference()
+        {
+            // Arrange
+            var action = new ActionBuilder()
+               .WithActionType(EActionType.TemplatedEmail)
+               .WithTo("test@abc.com")
+               .WithTemplateId("123")
+               .WithProvider("Notify")
+               .WithCaseReference(true)
+               .Build();
+
+            _mockActionHelper.Setup(_ => _.GetEmailToAddresses(It.IsAny<IAction>(), It.IsAny<FormAnswers>()))
+                .Returns("test@testemail.com");
+
+            var cacheData = new FormAnswers
+            {
+                CaseReference = "test-ref",
+                Path = "page-one",
+                Pages = new List<PageAnswers>()
+                {
+                    new()
+                    {
+                        Answers = new List<Answers>
+                        {
+                            new()
+                            {
+                                QuestionId = "firstname",
+                                Response = "test"
+                            }
+                        },
+                        PageSlug = "page-one"
+                    }
+                }
+            };
+            _mockDistributedCache.Setup(_ => _.GetString(It.IsAny<string>())).Returns(JsonConvert.SerializeObject(cacheData));
+
+            var personalisationSent = new Dictionary<string, dynamic>();
+
+            _mockTemplatedEmailProvider
+                .Setup(_ => _.SendEmailAsync(It.IsAny<string>(), It.IsAny<string>(),
+                    It.IsAny<Dictionary<string, dynamic>>()))
+                .Callback<string, string, Dictionary<string, dynamic>>((emailAddress, templateId, personalisation) => personalisationSent = personalisation);
+
+            // Act
+            _templatedEmailService.ProcessTemplatedEmail(new List<IAction> { action });
+
+            // Assert
+            Assert.Equal(personalisationSent["reference"], "test-ref");
+            Assert.Equal("1", personalisationSent.Count.ToString());
```

Comment:
u dont need to ToString this check, also there is an Assert on collection for a single item

Assert.Single(personalisationSent), this will check the collection contains a single item